deploy_snapshot_and_javadoc.sh

```diff
@@ -24,4 +25,26 @@ else
   echo "Deploying snapshot..."
   ./gradlew clean uploadArchives
   echo "Snapshot deployed!"
+
+  echo "Uploading javadoc..."
+
+  ./gradlew aggregatedJavadocs
+
+  {
+    set +x
+    cd $HOME
```

Comment:
This script tries to operate on the `~/javadoc` directory of whoever runs it, even if that directory has nothing to do with WALA. If that directory already exists, then hopefully the script will fail and stop at the `git clone` step, but this all makes me very nervous. Even if this script should only be run within an Travis CI job, the mere possibility of running `git rm -rf --quiet *` on a directory that might be outside of the WALA tree seems like an unnecessary risk.

Instead of working under `$HOME`, let’s change the script to work entirely within the `build` subdirectory of the current working directory when the script was launched. That will be the top-level `build` subdirectory of the WALA tree itself, which is where Gradle naturally puts all built artifacts. For example, we could check out the `wala/javadoc` project into `build/javadoc-for-gh-pages` and avoid any risk of damaging user files outside of the WALA tree.

Another option would be to simply use the `build/docs/javadoc` directory that the `aggregatedJavadocs` target will already have created. That directory is already exactly what we want to commit as the new `wala/javadoc`. The only thing that `build/docs/javadoc` is missing is Git metadata to treat it as a repository. So let’s change the script to add that, as in:

``` sh
# disclaimer: I have not actually tested this proposed sequence of commands!

./gradlew aggregatedJavadocs

cd build/docs/javadoc
git init
git commit -m "Latest javadoc on successful travis build $TRAVIS_BUILD_NUMBER auto-pushed to gh-pages" .  # don't overlook the important "." at the end of this line!

set +x
git remote add origin https://${GH_TOKEN}@github.com/wala/javadoc
set -x
git push --force
```

This avoids doing a `git clone` that we will simply blow away. On the other hand, `--force` is never to be taken lightly.